### DISCUSSION TITLE
Any way to move controlnet from "extensions-builtin" to the regular "extensions" folder?

### DISCUSSION THREAD

**Discussion by Sacred-Reputation** [vladmandic/sdnext#1837] [General]
see this reddit thread:

https://old.reddit.com/r/StableDiffusion/comments/159clmc/wildcard_for_preset_controlnet_poses/

I'm working with an extension called "Unprompted" and it wants to call ControlNet - but it's looking in the \automatic\extensions folder instead of \automatic\extensions-builtin

The developer of the Unpromted extension is replying in that reddit thread.

Thanks!

**Comment by vladmandic** [comment]
short answer: no.
if extension depends on another extension, it should have a mechanism to determine where it is, not to assume any specific path. and its trivial.
```python
import modules.extensions
print([e.name, e.path, e.enabled for e in modules.extensions.extensions])
```

who says someone would not install controlnet under a different folder name?
this is just **bad code**.

**Reply by Sacred-Reputation** [reply]
extension author said on reddit that he'll include that in the next update. thanks!
